pkg/config/validation/validation.go

```diff
@@ -3208,6 +3208,11 @@ var ValidateServiceEntry = registerValidateFunc("ValidateServiceEntry",
 			if port.TargetPort != 0 {
 				errs = appendValidation(errs, ValidatePort(int(port.TargetPort)))
 			}
+			if len(serviceEntry.Addresses) == 0 {
+				if port.Protocol == "" || port.Protocol == "TCP" {
+					errs = appendValidation(errs, WrapWarning(fmt.Errorf("one or more protocol invalid in service entry with addresses empty")))
```

Comment:
"invalid" is incorrect here. It is valid - it is just probably a bad idea. 

Maybe we can copy the similar text from the analyzer?

Reply:
it's done.